tests/test_api.py

```diff
@@ -1817,6 +1817,52 @@ def test_mandatory_conditions(self) -> None:
         )
         assert "deleted = 0" in result["sql"] or "equals(deleted, 0)" in result["sql"]
 
+    def test_hierarchical_hashes_array_slice(self) -> None:
+        result = json.loads(
+            self.post(
+                json.dumps(
+                    {
+                        "project": 1,
+                        "granularity": 3600,
+                        "selected_columns": [
+                            ["arraySlice", ["hierarchical_hashes", 0, 2]]
+                        ],
+                        "from_date": self.base_time.isoformat(),
+                        "to_date": (
+                            self.base_time + timedelta(minutes=self.minutes)
+                        ).isoformat(),
+                    }
+                ),
+            ).data
+        )
+
```

Comment:
You should also assert that these return a status code of 200. The SQL can be in the payload but the query be invalid.

Reply:
I'm getting errors that I cannot comprehend and that I feel are outside of the scope of this PR

Reply:
Let me pull your branch. Maybe I can figure it out.

Reply:
I modified the test to look like this:
```
response = self.post(
            json.dumps(
                {
                    "project": 1,
                    "granularity": 3600,
                    "selected_columns": [["arraySlice", ["hierarchical_hashes", 0, 2]]],
                    "from_date": self.base_time.isoformat(),
                    "to_date": (
                        self.base_time + timedelta(minutes=self.minutes)
                    ).isoformat(),
                }
            ),
        )
        assert response.status_code == 200
        result = json.loads(response.data)
        assert result["sql"].startswith(
            "SELECT arrayMap((x -> replaceAll(toString(x), '-', '')), arraySlice(hierarchical_hashes, 0, 2))"
        )
```

This passes for me when I run it. Your failures might be a local issue? Also these tests will fail if they run on events. @lynnagara Is this test always going to pass or will this fail for other environments (single tenant)?

Reply:
ok I'm pushing what I have, just not sure if it runs

Reply:
These tests only ever run on errors now